core/src/main/java/org/elasticsearch/cluster/block/ClusterBlocks.java

```diff
@@ -203,6 +203,29 @@ public ClusterBlockException indicesBlockedException(ClusterBlockLevel level, St
         return new ClusterBlockException(unmodifiableSet(blocks.collect(toSet())));
     }
 
+    public ClusterBlockException indicesAllowReleaseResources(String[] indices) {
+        boolean indexIsBlocked = false;
+        for (String index : indices) {
+            if (indexBlocked(ClusterBlockLevel.METADATA_WRITE, index)) {
+                indexIsBlocked = true;
+            }
+        }
+        if (globalBlocked(ClusterBlockLevel.METADATA_WRITE) == false && indexIsBlocked == false) {
+            return null;
+        }
+        Function<String, Stream<ClusterBlock>> blocksForIndexAtLevel = index -> blocksForIndex(ClusterBlockLevel.METADATA_WRITE, index)
```

Comment:
why do we need the first part of the method to shortcut this second half in the case we have no blocks? can't we make do with the second part only?